leonardnjura/propertypro-lyte
# #167392339 Setup tests with mocha and chai

#### What does this PR do?
- organizes models, migrations and environments with npx
- sets up mocha and chai for tests
- adds user route tests
- linting in builds with eslint and prettier
- pre-commit hooks with husky
- integrates travis CI and coveralls
- adds build and test coverage badges

#### Description of Task to be completed?


#### How should this be manually tested?
- Git `clone` repo and `cd` into this branch
- Setup Postgres
- Setup `.env` with example given. Ensure `NODE_ENV=test`
- `$ yarn install`
- `$ npm run server`
- `$ npm run lint`
- `$ npm run test`


#### What are the relevant pivotal tracker stories?
[#167392339](https://www.pivotaltracker.com/story/show/167392339)

#### Screenshots
![Screenshot 2019-07-21 at 10 57 10](https://user-images.githubusercontent.com/39657549/61588558-588bf980-aba6-11e9-98b9-fae3ff421a58.png)

![Screenshot 2019-07-21 at 10 59 08](https://user-images.githubusercontent.com/39657549/61588573-a3a60c80-aba6-11e9-881f-464ed74fd14d.png)
